File: samples/nodejs-typescript-food-catalog/src/functions/http-content.ts
import { app, HttpRequest, InvocationContext } from "@azure/functions";
import { CrawlType } from "../common/ContentMessage";
import { startCrawl } from "../common/queueClient";
import { streamToJson } from "../common/utils";

interface CrawlRequest {
  crawlType?: CrawlType;
}

app.http('content', {
  methods: ['POST'],
  route: 'crawl',
  handler: async (request: HttpRequest, context: InvocationContext) => {
    const body: CrawlRequest = await streamToJson(request.body);

    if (!body.crawlType || !['full', 'incremental'].includes(body.crawlType)) {
      return {
        status: 400
      }
    }

    context.log(`Enqueuing crawl request for ${body.crawlType}...`);
    startCrawl(body.crawlType);

    return {
      status: 202
    }
  }
})

File: samples/dotnet-csharp-graphdocs-ttk/Functions/HttpContent.cs
using Azure.Storage.Queues;
using GraphDocsConnector.Messages;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace GraphDocsConnector.Functions
{
    public class CrawlRequest
    {
        public CrawlType? CrawlType { get; set; }
    }

    public class HttpContent
    {
        private readonly ILogger<HttpContent> _logger;
        private readonly QueueClient _queueContentClient;

        public HttpContent(QueueClient queueClient, ILogger<HttpContent> logger)
        {
            _queueContentClient = queueClient;
            _logger = logger;
        }

        [Function("crawl")]
        public IActionResult Run([HttpTrigger(AuthorizationLevel.Anonymous, "post")] HttpRequest req, [FromBody] CrawlRequest crawlRequest)
        {
            if (crawlRequest.CrawlType is null)
            {
                return new BadRequestResult();
            }

            _logger.LogInformation($"Enqueuing crawl request for {crawlRequest.CrawlType}...");
            Queue.StartCrawl(_queueContentClient, crawlRequest.CrawlType.Value);

            return new AcceptedResult();
        }
    }
}
